src/main/java/org/jabref/preferences/PreferencesService.java

```diff
@@ -278,6 +279,10 @@
 
     void storeShouldAutosave(boolean shouldAutosave);
 
+    FileLinkPreferences getFileLinkPreferences();
+
+    void storeFileDirforDatabase(List<Path> dirs);
+
```

Comment:
Then I canot set this https://github.com/JabRef/jabref/pull/7636/files#diff-c9bd7e38f9d2f4a7eff486f337db736d6108369bb482c194ca8281ccfd61b93aL105

Reply:
Yes. This setter is no fix for the hack, but a disguise.

Reply:
Keep it ugly, until it's fixed. 😏

Reply:
Sure I would, but it needs to be present in the preferences Service, because we need to get rid of the Globals for the test, as we cannot mock it.

Reply:
I would vote to create a follow up PR to fix this

Reply:
Yes, a follow-up is most urgent needed, but there was a reason for that hack, I'm afraid, there is no easy fix for this. See declaration of this:
```
    // The following field is used as a global variable during the export of a database.
    // By setting this field to the path of the database's default file directory, formatters
    // that should resolve external file paths can access this field. This is an ugly hack
    // to solve the problem of formatters not having access to any context except for the
    // string to be formatted and possible formatter arguments.
    public List<Path> fileDirForDatabase;
```

Reply:
But to say that again with most deep concern: The proposed change, to put these two methods into the preferencesService shall not happen!

Reply:
After careful consideration: @Siedlerchr and me think it would be right, to keep the changes as SiedlerChr added, since extracting this mess of a temp variable in the preferences totally exploding this PR. This calls for a distinct one. Good luck, @Siedlerchr  😆